mgmtsystem/google.py

```diff
@@ -0,0 +1,312 @@
+
+    # Get external IP (ephemeral)
+    def current_ip_address(self, vm_name):
+        return self.vm_status(vm_name)['natIP']
+
+    def deploy_template(self, template, *args, **kwargs):
+        raise NotImplementedError('deploy_template not implemented.')
+
+    def disconnect(self):
+        raise NotImplementedError('disconnect not implemented.')
+
+    def does_vm_exist(self, instance_name):
+        try:
+            self._find_instance_by_name(instance_name)
+            return True
+        except Exception:
+            return False
+
+    def get_ip_address(self, vm_name):
+        return self.current_ip_address(vm_name)
+
+    def info(self):
+        raise NotImplementedError('info not implemented.')
+
+    def is_vm_running(self, vm_name):
+        return self.vm_status(vm_name) in self.states['running']
+
+    def is_vm_stopped(self, vm_name):
+        return self.vm_status(vm_name) in self.states['stopped']
+
+    def is_vm_suspended(self, vm_name):
+        raise NotImplementedError('is_vm_suspended not implemented.')
+
+    # These methods indicate if the vm is in the process of stopping or starting
+    def is_vm_stopping(self, vm_name):
+        return self.vm_status(vm_name) in self.states['stopping']
+
+    def is_vm_starting(self, vm_name):
+        return self.vm_status(vm_name) in self.states['starting']
+
+    def list_flavor(self):
+        raise NotImplementedError('list_flavor not implemented.')
+
+    def list_template(self):
+        raise NotImplementedError('list_template not implemented.')
+
+    def remove_host_from_cluster(self, hostname):
+        raise NotImplementedError('remove_host_from_cluster not implemented.')
+
+    def suspend_vm(self, vm_name):
+        raise NotImplementedError('suspend_vm not implemented.')
+
+    def vm_status(self, vm_name):
+        if self.does_vm_exist(vm_name):
+            return self._find_instance_by_name(vm_name)['status']
+        return None
+
+    def wait_vm_running(self, vm_name, num_sec=360):
+        self.logger.info("Waiting for instance {} to change status to ACTIVE".format(vm_name))
+        wait_for(self.is_vm_running, [vm_name], num_sec=num_sec)
+
+    def wait_vm_stopped(self, vm_name, num_sec=360):
+        self.logger.info("Waiting for instance {} to change status to TERMINATED".format(vm_name))
+        wait_for(self.is_vm_stopped, [vm_name], num_sec=num_sec)
+
+    def wait_vm_suspended(self, vm_name, num_sec):
+        raise NotImplementedError('wait_vm_suspended not implemented.')
+
+    def all_vms(self):
+        result = []
+        for vm in self._get_all_instances()['items']:
+            if (vm['id'] and vm['name'] and vm['status'] and
+                    vm['networkInterfaces'][0]['accessConfigs'][0]['natIP']):
+
+                result.append(VMInfo(
+                    vm['id'],
+                    vm['name'],
+                    vm['status'],
+                    vm['networkInterfaces'][0]['accessConfigs'][0]['natIP'],
```

Comment:
An important question :) does this return None in case the VM does not have an IP? Or it can't happen?

*/me gce noob*